### DISCUSSION TITLE
변수의 생명주기란 무엇인가요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#56] [전역 변수의 문제점]
- 변수의 생명주기란 무엇인가요?

**Comment by publdaze** [comment]
> - 변수의 생명주기란 무엇인가요?
- 변수가 생성되고 소멸될 때까지, 즉 변수의 메모리 공간이 확보된 시점부터 메모리 공간이 해제된 시점까지를 의미
- 전역 변수는 전역 객체의 프로퍼티가 되므로 전역 객체의 생명주기와 같고
- 지역 변수는 함수의 생명주기와 같음

**Comment by pipisebastian** [comment]
> 변수의 생명주기란 무엇인가요?

- 변수가 메모리에 할당되고, 해제될때까지의 시간입니다. 함수가 호출될때, 호이스팅이 일어나 변수가 선언되고 메모리에 할당됩니다. 그 후 함수가 종료될때까지의 시간입니다.
- 전역변수의 경우, 전역객체의 생명주기와 같으며, 지역변수의 경우 함수의 생명주기와 같습니다.
